ats/pkb/pkb.py

```diff
@@ -92,16 +139,40 @@ def on_node_enter(node: nodes.ASTNode, context: dict):
                         calls[callee] = set()
                     calls[callee].add(caller)
 
-        dfs(tree, on_node_enter=on_node_enter)
-
+        def on_node_exit(node: nodes.ASTNode, context):
+            if isinstance(node, nodes.VariableNode):
+                if node.parent.variable == node:
+                    modifies[node.name] = list(
+                        set(proc_stmt_stack + modifies[node.name])
+                    )
+                    if proc_stmt_stack[0] in proc_parents:
+                        modifies[node.name] = list(
+                            set(proc_parents[proc_stmt_stack[0]] + modifies[node.name])
+                        )
+                else:
+                    uses[node.name] = list(set(proc_stmt_stack + uses[node.name]))
+                    if proc_stmt_stack[0] in proc_parents:
+                        uses[node.name] = list(
+                            set(proc_parents[proc_stmt_stack[0]] + uses[node.name])
+                        )
```

Comment:
Co to robi? nie widze komentarzy wiec nie czytam kodu nawet

Reply:
w tym miejscu rozszerzamy rodziców procedury o innych rodziców procedury w kolejności jaką ustaliliśmy w call_order

Reply:
dodaj komentarze